src/SqlPersistence/Config/SqlPersistenceConfig_Connection.cs

```diff
@@ -15,24 +16,70 @@ public static void ConnectionBuilder(this PersistenceExtensions<SqlPersistence>
         {
             Guard.AgainstNull(nameof(configuration), configuration);
             Guard.AgainstNull(nameof(connectionBuilder), connectionBuilder);
+
             configuration.GetSettings()
-                .Set("SqlPersistence.ConnectionBuilder", connectionBuilder);
+                .Set("SqlPersistence.ConnectionManager", new SingleTenantConnectionManager(connectionBuilder));
+        }
+
+        /// <summary>
+        /// Configures how <see cref="DbConnection"/>s are constructed, allowing for selecting a different database per tenant in a multi-tenant system.
+        /// </summary>
+        /// <param name="configuration"></param>
+        /// <param name="tenantIdHeaderName"></param>
+        /// <param name="buildConnectionFromTenantData"></param>
+        public static void MultiTenantConnectionBuilder(this PersistenceExtensions<SqlPersistence> configuration, string tenantIdHeaderName, Func<string, DbConnection> buildConnectionFromTenantData)
```

Comment:
My only concern with the configuration API here is: what if tenant information are a runtime evaluated thing based on more than one header? Or you want to transition for a header name to another one?

Reply:
@mauroservienti in that case you use the second overload (below)

Reply:
Right now is private

Reply:
@mauroservienti right. I didn't notice that.

Reply:
With the new refactorings, this is no longer a problem.